### ISSUE TITLE
Chat - Messages are delivered with long delay after changing avatar

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#17190]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to https://staging.new.expensify.com/
on 2 devices with 2 accounts
2. Send message from A to B and back
3. Change avatar (remove current picture or upload a new one) - tester used Web
4. Go back to chat and send more messages

## Expected Result:
Messages should be sent without any issues

## Actual Result:
Looks like messages are not sent. One can see that the other account is typing but no messages arrive from A to B and back. It seems to both account that messages are sent but they are not delivered. Sometimes messages are delivered later with delay, sometimes they are not delivered at all.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] Windows / Chrome
- [ ] MacOS / Desktop

**Version Number:** 1.2.97.2

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/230746188-c99950ad-f1d6-40c9-b960-555bf53ddf30.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @flodnv (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by francoisl** [comment]
This is a real issue but it's also happening in production. Going to remove the deploy blocker label so we can go ahead with the deploy.

https://user-images.githubusercontent.com/2229301/230957443-461c52ea-0798-4c75-855b-04b21ec591c5.mov

**Comment by flodnv** [comment]
Yikes! What's going on here?! 😅 

@francoisl any idea if this is internal or external?

**Comment by francoisl** [comment]
No idea sorry, I was mostly looking to see if we could unblock the NewDot deploy. I checked the JS console when I tested and didn't see any errors, so for now I'd look at the network tab and make sure the requests are sent properly after updating your profile picture. Also backend logs, to see if there are weird shenanigans going on with push notifications.

**Comment by MelvinBot** [comment]
@flodnv Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
@flodnv 6 days overdue. This is scarier than being forced to listen to Vogon poetry!

**Comment by flodnv** [comment]
I finally got around to testing this, and the problem is in fact not in the API! The problem is that the NewDot client app waits on UpdateUserAvatar to be done before sending any subsequent requests! I am not sure why we do that, maybe something related to sending requests in sync / offline mode? In any case, in this scenario it really does not seem necessary to me... 

Video demonstrating this here: 

https://user-images.githubusercontent.com/4164921/232892206-78f4e714-2a12-4d38-9967-da2f906d0c83.mov

**Comment by flodnv** [comment]
Discussed here: https://expensify.slack.com/archives/C01GTK53T8Q/p1681848506766119

This is more a feature than a bug. It is kind of a bug, but we won't fix for now.
